optaplanner-core/src/main/java/org/optaplanner/core/config/solver/SolverConfig.java

```diff
@@ -257,6 +293,37 @@ protected RandomFactory buildRandomFactory(EnvironmentMode environmentMode_) {
         return randomFactory;
     }
 
+    protected Integer resolveMoveThreadCount() {
+        int availableProcessorCount = Runtime.getRuntime().availableProcessors();
+        Integer resolvedMoveThreadCount;
+        if (moveThreadCount == null || moveThreadCount.equals(MOVE_THREAD_COUNT_NONE)) {
+            return null;
+        } else if (moveThreadCount.equals(MOVE_THREAD_COUNT_AUTO)) {
+            // Leave one for the Operating System and 1 for the solver thread, take the rest
+            resolvedMoveThreadCount = (availableProcessorCount - 2);
+            if (resolvedMoveThreadCount <= 1) {
+                // Fall back to single threaded solving with no move threads.
+                // Set moveThreadCount explicitly to 1 to deliberately test with 1 moveThread.
+                return null;
+            }
+        } else {
+            resolvedMoveThreadCount = ConfigUtils.resolveThreadPoolSizeScript(
+                    "moveThreadCount", moveThreadCount, MOVE_THREAD_COUNT_NONE, MOVE_THREAD_COUNT_AUTO);
+        }
+        if (resolvedMoveThreadCount < 1) {
+            throw new IllegalArgumentException("The moveThreadCount (" + moveThreadCount
+                    + ") resulted in a resolvedMoveThreadCount (" + resolvedMoveThreadCount
+                    + ") that is lower than 1.");
+        }
```

Comment:
Yes, it could. Findbugs or PMD might complain about this too.
But we might add other options later (other if else's) and this is a fail-fast check that happens after the number has been resolved. I am going to leave it as-is. Thanks for reporting though.